### PR TITLE
feature: support request timeout

### PR DESCRIPTION
**Pull request by morganxf**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,6 +6,7 @@
 # Folders
 _obj
 _test
+.idea/
```
Comment (SuperQ) [inline .gitignore]: This doesn't seem necessary for this change.

File: cmd/prom2json/main.go [hunk 2]
```diff
@@ -26,9 +27,12 @@ import (
 	"github.com/prometheus/prom2json"
 )
 
+const DefaultTimeout = 10 * time.Second
```
Comment (SuperQ) [inline cmd/prom2json/main.go]: I'm pretty sure this can be `defaultTimeout`, it doesn't need to be exported.

Comment (beorn7) [reply cmd/prom2json/main.go]: In different news, I think the default behavior should be "no timeout" (as it was before this change).

**Comment by morganxf** [comment]
ok
I will pull another request with no timeout by default.

**Comment by beorn7** [comment]
You also have to add the DCO, see https://github.com/prometheus/prometheus/wiki/DCO-signing

**Comment by beorn7** [comment]
Closing for lack of updates.

Also, `prom2json` can now read from stdin (soon to be released as 1.1.0), so you can call `curl` with all its fancy command line flags and then pipe the result into `prom2json`. I think that is much saner than gradually re-implementing `curl` within `prom2json`.

**Pull request closed without merging**